### ISSUE TITLE
:e!

### ISSUE THREAD

**Issue by polluks** [kyx0r/nextvi#23]
I read ":e!  reedit, discard changes".
Is your implementation POSIX conform?

**Comment by kyx0r** [comment]
https://0x0.st/oea9.txt
It is.
Where does it say that changes must be discarded?
Technically they would be discarded; however, since
we have buffers implemented (which is out of scope of this specification)
we open a new buffer instead. So if you think that the new buffer is the same as
the old, then it does discard the changes correctly. Or set bx to 1 so that
there are no extra buffers and it will work.

**Comment by kyx0r** [comment]
Also fun fact: notice that certain things are omitted in the presence of buffers.
For example, you can have some buffers marked as modified but if the current
buffer is marked as saved then :q will still work as it only checks only the current
buffer and not others. So you can do self harm if you are not careful, but in practice
it's probably too inconvenient anyway so I don't bother implementing it.

**Comment by polluks** [comment]
Aha, so you don't want to misuse `e` as `rew`?

**Comment by kyx0r** [comment]
It's just that by default e command opens the "unnamed" buffer, which is not the case in neatvi or the likes. Neatvi treats e as if it was e %. However, even if you use e % in nextvi, it will not perform the "discard changes" since e % will just switch to whatever buffer with % already exists. For this operation I use 1,$d|r|1 which does exactly the same. The problem is if I make it conform then it will be impossible to create more than 1 unnamed buffer. This is stupid, e should not take the role of r command, but this is what spec implies. I can't just make e! % work like either, since ! also implies autowrite so if aw is set it's going to re-read what it just wrote.

**Comment by kyx0r** [comment]
@polluks Good news, I've made a decision to at least be partially compliant with this. But only for e! and not e. For compatibility with old versions I need to keep e working for opening unnamed buffers.

BTW there was another idea, is to reread only when the file was changed and if it wasn't use e and e! to open unnamed buffers. However I think it's a bit loose design, and more or less unpredictable behavior since it would rely on the file time stamp.
